chain/chunks/src/lib.rs

```diff
@@ -637,24 +643,25 @@ impl ShardsManager {
 
         if let Some(header_head) = header_head {
             let fetch_from_archival = self.runtime_adapter
-                .chunk_needs_to_be_fetched_from_archival(&parent_hash, &header_head.last_block_hash).unwrap_or_else(|err| {
+                .chunk_needs_to_be_fetched_from_archival(&ancestor_hash, &header_head.last_block_hash).unwrap_or_else(|err| {
                 error!(target: "chunks", "Error during requesting partial encoded chunk. Cannot determine whether to request from an archival node, defaulting to not: {}", err);
                 false
             });
-            let old_block = header_head.last_block_hash != parent_hash
-                && header_head.prev_block_hash != parent_hash;
+            let old_block = header_head.last_block_hash != prev_block_hash
+                && header_head.prev_block_hash != prev_block_hash;
 
-            // If the protocol version is later than the one where chunk forwarding was introduced,
+            // If chunk forwarding is enabled,
             // we purposely do not send chunk request messages right away for new blocks. Such requests
             // will eventually be sent because of the `resend_chunk_requests` loop. However,
             // we want to give some time for any `PartialEncodedChunkForward` messages to arrive
             // before we send requests.
             let is_chunk_forwarding_enabled =
                 checked_feature!("stable", ForwardChunkParts, protocol_version);
+            let is_chunk_forwarding_enabled = false;
```

Comment:
It's a temporary hack here. The issue is that chunk parts won't be forwarded to non-validators, so the code here should check whether the node is a validator or not and only delay the request if it's a validator.

Reply:
waiting for https://github.com/near/nearcore/pull/5360 to be merged